README.adoc

```diff
@@ -272,6 +272,208 @@ cd applications/sink/log-sink/apps/log-sink-rabbit
+
+Then inside your application, you can call `mySupplier().get()` and retrieve the data.
+
+===== Supplying data using reactive patterns.
+
+Below, we are discussing two patterns using which you can write reactive suppliers.
+
+a. Let's say that you have a Spring Integration based message driven adapter that you want to recieva data as soon as it is available.
+You want to produce this data through a supplier for your clients.
+You can use the following primitives as a starting point for that.
+We use the TCP adapter as an example here:
+
+```
+
+@Bean
+public Supplier<Flux<Message<?>>> tcpSupplier() {
+    return () -> Flux.from(output())
+            .doOnSubscribe(subscription -> adapter().start());
+}
+
+@Bean
+public FluxMessageChannel output() {
+    return new FluxMessageChannel();
+}
+
+@Bean
+public TcpReceivingChannelAdapter adapter() {
+    TcpReceivingChannelAdapter adapter = new TcpReceivingChannelAdapter();
+    adapter.setOutputChannel(output());
+    adapter.setAutoStartup(false);
+    ...
+    return adapter;
+}
+```
+
+b. If you have a custom mechanism (non Spring Integration) and want to produce data reactively, you can follow the blueprint below.
+
+```
+@Bean
+public Publisher<Message<byte[]>> myPublisher() {
+    ...
+}
+
+@Bean
+public Supplier<Flux<Message<?>>> mySupplier() {
+    return () -> Flux.from(myPublisher())
+            .doOnSubscribe(subscription -> adapter().start());
+}
+```
+
+Please take a look at the various suppliers provided link:functions/supplier[here] more usage examples of these various types of suppliers.
+
+===== Common conventions used when writing a Supplier
+
+Following are the expected conventions when writing a new Supplier, taking Google Spanner as an example.
+
+* The name of the configuration class - `SpannerSupplierConfiguration`
+* The name of the supplier method - `spannerSupplier`.
+* `ConfigurationProperties` class is named as - `SpannerSupplierProperties`.
+
+==== Writing a new Consumer function
+
+Writing a consumer functions is more straightforward than writeing a `Supplier`.
+You can simple provide a `Consumer` bean as below.
+
+```
+@Bean
+public Consumer<String> myConsumer() {
+    return data -> sendDataT();
+}
+```
+
+Or if you are relying on some Spring Integration message handlers,
+
+```
+@Bean
+public Consumer<Message<?>> fileConsumer() {
+    return fileWritingMessageHandler()::handleMessage;
+}
+```
+
+link:functions/consumer[Here] are all the consumer functions we currently have.
+
+===== Common conventions used when writing a Consumer
+
+Following are the expected conventions when writing a new Consumer, taking Google Spanner as an example.
+
+* The name of the configuration class - `SpannerConsumeronfiguration`
+* The name of the supplier method - `spannerConsumer`.
+* `ConfigurationProperties` class is named as - `SpannerConsumerProperties`.
+
+==== Writing a new Function
```

Comment:
Maybe there are two separate but related topics:  1) What we expect for contributions, naming conventions, coding, testing and documentation standards, build procedure. The naming conventions are actually requirements for the Maven plugin.  2) How to write a function.

Reply:
1) Would not not include any code samples, nothing about implementation details.  just "look at the ones we already have".   
2) Would describe patterns, configuration properties, Spring vs non-Spring, etc. Maybe this could be under the docs in functions/